Remove duplication of code

fredmanglis opened the pull request:
The functions `get_trait_csv_sample_data` and `retrieve_publish_trait_data` in
the `gn3.db.traits` module ran mostly the same query, and thus had sections
that were duplicates of each other. In this commit, the
`retrieve_publish_trait_data` is reused in the `get_trait_csv_sample_data`
function to remove the duplication.

This commit also fixes some linting, and typing issues.

fredmanglis commented:
As requested by @arunisaac I have looked at the duplicate code in the two functions and removed the duplication, while also fixing some linting and typing issues.

@BonfaceKilz Please have a look in order to verify that my changes do not break the usage of the `gn3.db.traits.get_trait_csv_sample_data` function.

If everything is good from both of you, please feel free to merge this.

arunisaac commented:
The linting and typing issues are unrelated. So, I recommend splitting them
off into two separate commits.

Line 1 of the commit message could be a little more descriptive---something
like "Deduplicate PublishData query" or similar. Currently, it is not clear
what code is being deduplicated. Remember that git treats the first line of
the commit message as a summary in `git log --oneline` and many other
places. So, the first line should be able to more or less stand on its own.

Very minor nitpick: In many places (for example, line 90 of
`gn3/db/traits.py`), you break the line at an open parentheses. I feel that it
would be more readable to break the line at a function argument or
elsewhere. But, this could be just my lisp brain talking. So, feel free to
ignore.

Otherwise, LGTM! Thanks for working on this! I appreciate that you did this
even though it wasn't strictly necessary. :-)

fredmanglis committed:
Deduplicate query to fetch data for 'Publish' traits

The queries run in the `get_trait_csv_sample_data` and
`retrieve_publish_trait_data` functions in the `gn3.db.traits` module were
mostly similar. This commit changes that, by making the
`get_trait_csv_sample_data` function make use of the results from calling the
`retrieve_publish_trait_data` function.

fredmanglis committed:
Indent code correctly

Indent the code correctly.

fredmanglis committed:
Update return type. Remove duplicate import.

The function is a generator function, since it uses a `yield` statement, and
thus returns a generator object, that contains a tuple object. This fixes
that. We also remove a duplicate import.

fredmanglis commented:
@arunisaac I have done the changes requested: please have a look when you have a minute.

I will merge this ASAP to get it moving forward.

arunisaac commented:
> @arunisaac I have done the changes requested: please have a look when
> you have a minute.

Looks good, thanks!
